### DISCUSSION TITLE
Interaction failed for simple selector UI

### DISCUSSION THREAD

**Discussion by joejo-waywise** [Rapptz/discord.py#9926] [Q&A] [answered]
Below is a very simple selector implementation that is based on the sample dropdown.py in the repo. If I select any item in the dropdown, all I get is a error message in the Discord App that 'This interaction failed.` but I'm not sure why. Print debugging shows that the callback is never being called.

As a general question: What is the best way to go about debugging something like this?

Any help is appreciated.

```py
class MySelector(discord.ui.Select):

    def __init__(self):
        options = [
            discord.SelectOption(label='One'),
            discord.SelectOption(label='Two'),
            discord.SelectOption(label='Three'),
        ]

        super().__init__(placeholder="Select an option", min_values=1, max_values=1, options=options)

    async def callback(self, interaction: discord.Interaction):
        print('callback handler')
        await interaction.response.edit_message(content=f'You selected: {self.values[0]}')

class MyView(discord.ui.View):
    def __init__(self):
        super().__init__()
        self.add_item(MySelector())


class TestCog(commands.Cog):

    def __init__(self, bot):
        self.bot = bot

    @commands.command()
    async def select(self, ctx):
        print('command handler')
        view = MyView()
        await ctx.send('Select an option', view=view)
```

This sample is dealing with a cog, but that cog is installed in the bot using `add_item` at startup time and never unloaded or reloaded.

**Comment by Soheab** [comment]
Can you please fix your attempted codeblocks... they should be like:
\`\`\`py
CODE HERE
\`\`\`
^^^ 3 ticks

**Reply by joejo-waywise** [reply]
Done, sorry

**Comment by ImLimiteds** [comment]


**Reply by Rapptz** [reply]
This is completely incorrect.

**Reply by joejo-waywise** [reply]
Why is it incorrect?

**Comment by Rapptz** [comment]
The code as-is should work fine. However, by default the library only listens to components that are live during the bot's execution. So if these dropdowns are older than your startup the bot is unaware of them. If this is the case you need to use either [`DynamicItem`](https://github.com/Rapptz/discord.py/blob/master/examples/views/dynamic_counter.py) or a [persistent view](https://github.com/Rapptz/discord.py/blob/master/examples/views/persistent.py).

**Reply by joejo-waywise** [reply]
Well, the view is only constructed when a command is handled by the cog. The Cog is installed in the bot at startup. I don't think the bot can handle a command unless it's running, so I am going to assume that that is not a current issue with the code (though I do see that would be an issue for a cog that is loaded/unloaded dynamically).

If this is an issue, then it might be good to add a sample that clarifies this. AFAICT there isn't even a cog sample much less one that takes this into consideration and deals with it.

All that aside, the fact that it isn't working fine would seem to be an issue then. I will try to use one or the other of your suggestions, but those would seem to be contra-indicated in this example.

**Reply by Rapptz** [reply]
Cog unloading and reloading doesn't affect this. Only what I mentioned does. Note that views only last for the timeout before the library stops responding to them too, which by default (like your example) is 3 minutes. If you want it to be higher you can set the timeout higher but for "infinite timeout" you want it to be a dynamic item or a persistent view.

**Reply by joejo-waywise** [reply]
I'm not taking 3 minutes to respond or handle the UI. 

I did try DynamicItem, but I don't think that it works with Select:

```py
Traceback (most recent call last):
  File "/home/waywise-dev/project/src/discord_bot/app.py", line 8, in <module>
    from test_cog import TestCog
  File "/home/waywise-dev/project/src/discord_bot/test_cog.py", line 16, in <module>
    class GameSelector(discord.ui.DynamicItem(discord.ui.Select), template=r""):
  File "/home/waywise-dev/.local/lib/python3.10/site-packages/discord/ui/dynamic.py", line 110, in __init__
    if not self.item.is_dispatchable():
TypeError: BaseSelect.is_dispatchable() missing 1 required positional argument: 'self'
```

**Reply by Rapptz** [reply]
That's not how you do inheritance. Please double check the example, it's using brackets instead of parentheses for that portion for type checking purposes.

It's not 3 minutes to respond, it's 3 minutes since the view is sent.

**Reply by joejo-waywise** [reply, marked as answer]
>  it's using brackets instead of parentheses for that portion for type checking purposes.

doh.

**Reply by joejo-waywise** [reply]
Ok, neither option worked so I went spelunking around in different things. Turns out there was an interaction endpoint specified in the bot from an earlier investigation, and that endpoint didn't exist. deleting that made everything work.

Thanks, all.
